src/mlpack/core/dists/gamma_distribution.cpp

```diff
@@ -55,36 +57,63 @@ void GammaDistribution::Train(const arma::mat& rdata, const double tol)
   // Treat each dimension (i.e. row) independently.
   for (size_t row = 0; row < rdata.n_rows; ++row)
   {
-    // Statistics for this row.
+    // Statistics for this Vrow.
     const double meanLogx = meanLogxVec(row);
     const double meanx = meanxVec(row);
     const double logMeanx = logMeanxVec(row);
 
-    // Starting point for Generalized Newton.
-    double aEst = 0.5 / (logMeanx - meanLogx);
-    double aOld;
+    // Use the statistics-only function to fit this dimension.
+    Train(logMeanx, meanLogx, meanx, tol);
+
+    // The function above modifies the state of this object. Get the parameters
+    // it fit. (This is not very good design...).
+    tempAlpha(row) = alpha(0);
+    tempBeta(row) = beta(0);
+  }
+  alpha = tempAlpha;
+  beta = tempBeta;
+
+}
+
+// Fits an alpha and beta parameter to each dimension of the data.
+void GammaDistribution::Train(const double logMeanx, const double meanLogx,
+                              const double meanx,
+                              const double tol)
```

Comment:
You're right - I was concerned with my design using temps and having two functions changing the vectors continuously. What you propose is cleaner. Thanks :)